TwitchDownloaderCLI/Modes/UpdateHandler.cs

```diff
@@ -0,0 +1,207 @@
+            {
+                // TODO: Change to 'linux-musl-x64' when .NET 8+
+                if (RuntimeInformation.RuntimeIdentifier.Contains("musl"))
+                {
+                    return string.Format(origPackageName, "LinuxAlpine-x64");
+                } 
+                else
+                {
+                    return arch switch
+                    {
+                        Architecture.X64 => string.Format(origPackageName, "Linux-x64"),
+                        Architecture.Arm => string.Format(origPackageName, "LinuxArm"),
+                        Architecture.Arm64 => string.Format(origPackageName, "LinuxArm64"),
+                        _ => string.Empty
+                    };
+                }    
+            }
+
+            return string.Empty;
+        }
+
+        private static async Task AutoUpdate(string url, CliTaskProgress progress)
+        {
+            var currentExePath = Environment.ProcessPath;
+            var oldExePath = currentExePath + ".bak";
+            var updateDir = Path.GetDirectoryName(currentExePath)!;
+            var archivePath = Path.Combine(updateDir, url.Split("/").Last());
+
+            if (string.IsNullOrEmpty(currentExePath))
+            {
+                progress.LogError("Current executable path is null or empty!");
+                return;
+            }
+
+            if (string.IsNullOrEmpty(updateDir))
+            {
+                progress.LogError("Update directory is null or empty!");
+                return;
+            }
+
+            progress.SetTemplateStatus("Downloading Update {0}%", 0);
+
+            using var response = await _client.GetAsync(url, HttpCompletionOption.ResponseHeadersRead);
+            response.EnsureSuccessStatusCode();
+            await using var contentStream = await response.Content.ReadAsStreamAsync();
+
+            var archiveLength = response.Content.Headers.ContentLength;
+
+            // Create downloaded archive file from stream data
+            await using (var fs = new FileStream(archivePath, FileMode.OpenOrCreate))
+            {
+                await contentStream.ProgressCopyToAsync(fs, archiveLength, new Progress<StreamCopyProgress>(DownloadProgressHandler)).ConfigureAwait(false);
+            }
+
+            // Check for a previous update
+            if (File.Exists(oldExePath))
+            {
+                File.Delete(oldExePath);
+            }
+
+            // Rename current exe
+            File.Move(currentExePath!, oldExePath);
+
+            await using (var archiveFs = File.Open(archivePath, FileMode.Open, FileAccess.Read, FileShare.Read))
+            {
+                using var archive = new ZipArchive(archiveFs, ZipArchiveMode.Read);
+                foreach (var entry in archive.Entries)
+                {
+                    entry.ExtractToFile(Path.Combine(updateDir, entry.FullName), true);
+                }
+            }
```

Comment:
I ran the updater again and noticed extracting takes a while sometimes so I wrote a progress for that too.
```suggestion
            progress.SetTemplateStatus("Extracting Files {0}%", 0);

            await using (var archiveFs = File.Open(archivePath, FileMode.Open, FileAccess.Read, FileShare.Read))
            {
                using var archive = new ZipArchive(archiveFs, ZipArchiveMode.Read);

                var entryCount = archive.Entries.Count;
                var extracted = 0;

                foreach (var entry in archive.Entries)
                {
                    entry.ExtractToFile(Path.Combine(updateDir, entry.FullName), true);
                    extracted++;

                    var percent = (int)(extracted / (double)entryCount * 100);
                    progress.ReportProgress(percent);
                }
            }
```

Reply:
Added in commit [5e03a7d](https://github.com/lay295/TwitchDownloader/pull/1345/commits/5e03a7dc668451fd3cb54f6abc8c0645ac69424d)